### ISSUE TITLE
Swagger success response should be populated based on ResourceDoc.successResponseBody or by introspecting ResourceDoc.partialFunction return type

### ISSUE THREAD

**Issue by simonredfern** [OpenBankProject/OBP-API#31]
Currently in SwaggerJSONFactory.scala we have a place holder values for the response types:
Line 76:
      Map("200" -> ResponseObjectJson(Some("Success") , None), "400" -> ResponseObjectJson(Some("Error"), Some(ResponseObjectSchemaJson("#/definitions/Error"))))))

As an example, in API Methods we have lazy val getTransactionTypes which is returning a List of TransactionTypes. For this call, we should populate the success response (line 76 above) using the return type - either directly by introspecting ResourceDoc.partialFunction i.e. getTransactionTypes in this case or explitily in ResourceDocs.successResponseBody (maybe we need to change the type of successResponseBody?).

**Comment by simonredfern** [comment]
you can use getBanks instead

**Comment by marko-md** [comment]
In order to do the task we have to transform, for instance, next case class

**case class Bank(id: Long, name: String, email: Option[String])**

to next swagger output which represent schema object 

**definitions:
  bank:
    required:
      - id
      - name
    properties:
      id:
        type: integer
        format: int64
      name:
        type: string
      tag:
        type: string**
        
As far as I know it cannot be done directly by used lift-json. We have to develop a new function for that. After that we have to merge it with the current result through Json AST.
Idea is that we have to work only with existing case classes and avoid code duplication.

**Comment by simonredfern** [comment]
@marko-md Can we close this ticket now?

**Comment by marko-md** [comment]
As far as I remember it shouldn't be closed. In ResourceDoc parameter successResponseBody is type of JValue. When we try to extract some class from it, in order to produce swagger output, we have to explicitly type the name of that class. For example the current state is:
```scala
case u if u.apiFunction.contains("allBanks") => rd.successResponseBody.extract[BanksJSON]
case u if u.apiFunction.contains("bankById") => rd.successResponseBody.extract[BankJSON]
```
So we have to omit translation of case class to JValue. In that case it will be dynamically bound and swagger output will reflect automatically any change in ResourceDoc.

**Comment by hongwei1** [comment]
fixed in #511
